src/components/external-source/ExternalSourceManager.svelte

```diff
@@ -0,0 +1,1411 @@
+              tooltipContent="View plans this source is used in"
+            >
+              {#if selectedSourceLinkedDerivationGroupsPlans.length > 0}
+                {#each selectedSourceLinkedDerivationGroupsPlans as linkedPlanDerivationGroup}
+                  <i>
+                    <a href="{base}/plans/{linkedPlanDerivationGroup.plan_id}">
+                      {$plans.find(plan => {
+                        return linkedPlanDerivationGroup.plan_id === plan.id;
+                      })?.name}
+                    </a></i
+                  >
+                {/each}
+              {:else}
+                <i>Not used in any plans</i>
+              {/if}
+            </Collapse>
+          </div>
+          <button
+            class="st-button danger w-100"
+            style="margin-bottom:auto;"
+            use:permissionHandler={{
+              hasPermission: hasDeletePermission,
+              permissionError: deletePermissionError,
+            }}
+            on:click|stopPropagation={async () => onDeleteExternalSource(selectedSource)}
+          >
+            Delete external source
+          </button>
+        </div>
+      {:else}
+        <form
+          on:submit|preventDefault={onFormSubmit}
+          on:reset={() => {
+            parsed = undefined;
+            $createExternalSourceError = null;
+            $createExternalSourceTypeError = null;
+            $createDerivationGroupError = null;
+            $parsingError = null;
+            isDerivationGroupFieldDisabled = true;
+          }}
+        >
+          <AlertError class="m-2" error={$createExternalSourceError} />
+          <AlertError class="m-2" error={$createExternalSourceTypeError} />
+          <AlertError class="m-2" error={$createDerivationGroupError} />
+          <AlertError class="m-2" error={$parsingError} />
+          <div id="file-upload-field">
+            <fieldset style="width:100%">
+              <label for="file">Source File</label>
+              <input
+                class="w-100"
+                name="file"
+                required
+                type="file"
+                bind:files
+                use:permissionHandler={{
+                  hasPermission: hasCreatePermission,
+                  permissionError: createPermissionError,
+                }}
+              />
+            </fieldset>
+
+            <fieldset id="file-upload-fieldset">
+              <button
+                disabled={!parsed}
+                class="st-button w-100"
+                type="submit"
+                use:permissionHandler={{
+                  hasPermission: hasCreatePermission,
+                  permissionError: createPermissionError,
+                }}
+              >
+                {$creatingExternalSource ? 'Uploading...' : 'Upload'}
+              </button>
+              {#if parsed}
+                <div style="padding-top:10px">
+                  <button class="st-button w-100" type="reset">Dismiss</button>
```

Comment:
Would be good to inline these two buttons and give the "dismiss" one the `secondary` class.
<img width="505" alt="image" src="https://github.com/user-attachments/assets/559d3bd5-2708-4934-a021-ec191491caba">

Reply:
Is this what you were envisioning for inlining the buttons?
![image](https://github.com/user-attachments/assets/1e2b0628-ca9e-4453-be4d-a169cac5d100)
Code below, note this also implements the change in [this comment](https://github.com/NASA-AMMOS/aerie-ui/pull/1396#discussion_r1733303000)
```html
<div id="file-upload-field">
  <fieldset style:flex={1}>
    <div style:display="grid">
      <label for="file">Source File</label>
      <input
        class="w-100"
        name="file"
        required
        type="file"
        bind:files
        use:permissionHandler={{
          hasPermission: hasCreatePermission,
          permissionError: createPermissionError,
        }}
      />
    </div>
    <div style:padding-top="10px">
      {#if parsed}
        <button
          disabled={!parsed}
          class="st-button w-50"
          type="submit"
          style:display="inline"
          use:permissionHandler={{
            hasPermission: hasCreatePermission,
            permissionError: createPermissionError,
          }}
        >
          {$creatingExternalSource ? 'Uploading...' : 'Upload'}
        </button>
        <button class="st-button secondary w-50" style:display="inline" type="reset">Dismiss</button>
      {:else}
        <button
          disabled={!parsed}
          class="st-button w-100"
          type="submit"
          style:display="inline"
          use:permissionHandler={{
            hasPermission: hasCreatePermission,
            permissionError: createPermissionError,
          }}
        >
          {$creatingExternalSource ? 'Uploading...' : 'Upload'}
        </button>
      {/if}
    </div>
  </fieldset>
</div>
```